### PR TITLE
Feat: Adds `DiscountBadge` component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

- Creates `DiscountBadge` component on @faststore/components
- Creates `DiscountBadge` styles on @faststore/ui
- Adds `DiscountBadge` documentation page on Docusaurus

## How it works?

A custom `Badge`  that display product's discounts.

## How to test it?

run yarn start in app/docs folder and checks the Icon component or test it using the doc [preview link.](https://faststore-git-feat-discount-badge-component-fs-623-faststore.vercel.app/reference/ui/molecules/DiscountBadge)

### Starters Deploy Preview

[Feat: Replaces DiscountBadge FSUI #313](https://github.com/vtex-sites/nextjs.store/pull/313)

### CODE REVIEW

File: packages/ui/src/components/molecules/DiscountBadge/DiscountBadge.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+export const Template = (args) => (
+  <DiscountBadge id="checkbox-field" listPrice={65} spotPrice={40} />
+)
+
+<header>
+
+# DiscountBadge
+
+CheckboxField is composed of a Checkbox input and its corresponding Label. It allows users to select one option from a set.
```
Comment (eduardoformiga) [inline packages/ui/src/components/molecules/DiscountBadge/DiscountBadge.stories.mdx:15]: Left behind?
```suggestion
A custom `Badge` that display product's discounts.
```

File: packages/ui/src/components/molecules/DiscountBadge/DiscountBadge.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs'
+
+import { DiscountBadge } from '@faststore/components'
+
+<Meta title="Molecules/DiscountBadge" component={DiscountBadge} argTypes={{}} />
+
+export const Template = (args) => (
+  <DiscountBadge id="checkbox-field" listPrice={65} spotPrice={40} />
```
Comment (eduardoformiga) [inline packages/ui/src/components/molecules/DiscountBadge/DiscountBadge.stories.mdx:8]:
```suggestion
  <DiscountBadge id="discount-badge-id" listPrice={65} spotPrice={40} />
```

**Review by eduardoformiga — COMMENTED** [review]
[the preview](https://faststore-r08a0l5c0-faststore.vercel.app/reference/ui/molecules/DiscountBadge) looks amazing!

I left some minor comments.

File: packages/ui/src/components/molecules/DiscountBadge/DiscountBadge.stories.mdx (hunk 1 above)
Comment (hellofanny) [reply packages/ui/src/components/molecules/DiscountBadge/DiscountBadge.stories.mdx:15]: oopssss 🙈

File: apps/docs/docs/reference/ui/molecules/DiscountBadge.mdx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+import '@faststore/ui/src/components/molecules/DiscountBadge/styles.scss'
+```
+
+## Usage
+
+```tsx live
+<DiscountBadge listPrice={45} spotPrice={40} />
+```
+
+## Props
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/molecules/DiscountBadge.mdx:46]:
```suggestion
### Props
```

File: apps/docs/docs/reference/ui/molecules/DiscountBadge.mdx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+<PropsSection name="DiscountBadge" />
+
+## Variants
+
+### Low Discount
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/molecules/DiscountBadge.mdx:52]:
```suggestion

---

## Design Tokens

### Variants

#### Low Discount
```

File: apps/docs/docs/reference/ui/molecules/DiscountBadge.mdx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  />
+  <TokenRow
+    token="--fs-discount-badge-low-border-color"
+    value="var(--fs-badge-success-border-color)"
+    globalValue="var(--fs-color-success-bkg)"
+    isColor
+  />
+</TokenTable>
+
+### Medium Discount
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/molecules/DiscountBadge.mdx:79]:
```suggestion
#### Medium Discount
```

File: apps/docs/docs/reference/ui/molecules/DiscountBadge.mdx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+  />
+  <TokenRow
+    token="--fs-discount-badge-medium-border-color"
+    value="var(--fs-badge-warning-border-color)"
+    globalValue="var(--fs-color-warning-bkg)"
+    isColor
+  />
+</TokenTable>
+
+### High Discount
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/molecules/DiscountBadge.mdx:106]:
```suggestion
#### High Discount
```

File: apps/docs/docs/reference/ui/molecules/DiscountBadge.mdx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+  />
+  <TokenRow
+    token="--fs-discount-badge-high-border-color"
+    value="var(--fs-badge-danger-border-color)"
+    globalValue="var(--fs-color-danger-bkg)"
+    isColor
+  />
+</TokenTable>
+
+## Customization
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/molecules/DiscountBadge.mdx:133]:
```suggestion

---

## Customization
```

**Review by renatamottam — COMMENTED** [review]
Left a couple doc improvements :)

**Review by eduardoformiga — APPROVED** [review]

**Review by renatamottam — APPROVED** [review]

**Pull request merged**